theano/gof/op.py

```diff
@@ -783,6 +783,9 @@ def __new__(cls, *args, **kwargs):
         # this function exists to silently and transparently ensure that all
         # existing Ops get a _op_use_c_code attribute
         obj = object.__new__(cls)
+        # Ensuring all the Ops created henceforth has __props__
+        if not hasattr(obj, '__props__'):
+            raise
```

Comment:
While passing parameters, it becomes confusing and there are some errors at making call to `__call__`. I don't know if this could be fixed without requirement of `__props__` I would need @abergeron 's help. But with `__props__`,  I think i can fix it.